cheri.hpp

```diff
@@ -36,62 +39,66 @@ struct CheriStackIterator
 {
   void **location;
 
-  // Check if the pointer is pointing on the stack.
-  // Done by checking it aginst the bottom stack pointr's bounds.
-  bool is_stack(void *ptr)
-  {
-    return cheri_is_address_inbounds(bottom, cheri_getaddress(ptr));
-  }
-
   // Check if the pointer is executable. 
   // Done by using the permition bits.
   bool is_exec(void *ptr)
   {
-    return ((cheri_getperm(ptr) & 0b10) >> 1);
-  }
+    TRACE("%p %d\n", ptr, (cheri_getperm(ptr) != 0x6817d));
+    //inspect_pointer(*location);
+    return ((cheri_getperm(ptr) != 0x6817d) );
 
-  CheriStackIterator()
-  {
-    location = reinterpret_cast<void **>(bottom);
+  }
+  
+  // Skip all the things that arn't pointers.
+  // Also skip all the pointers we don't care about.  
+  void find_next() {
+    TRACE("%lu\n", cheri_getaddress(bottom) - cheri_getaddress(location));
+    while(!cheri_gettag(*location) or !cheri_getflags(*location) or is_exec(*location))  
+    {
+      if(location >= bottom) {
+        location = (void**)bottom;
+        return;
+      }
+      //TRACE("Looking for pointers: %p %p\n", location, *location);
```

Comment:
Commented code.

Reply:
Ah, this one specific case is the only part I decided to leave the commented debug code. Because with it there is too much output and it gets hard to discern what is going on, but if debugging of this exact part is needed it is usefull. I suppose I could have a VERBOSE macro?

Reply:
Personally I'm of the opinion that it's rarely a good idea to leave debugging code in. But if you do want to do so, then yes, a `VERBOSE` flag is OK.